cmd/zrepl/mgmt_conn.c

```diff
@@ -163,6 +186,43 @@ close_conn(uzfs_mgmt_conn_t *conn)
 	return (0);
 }
 
+/*
+ * Complete destruction of conn struct. conn list mtx must be held when calling
+ * this function.
+ * Close connection if still open, remove conn from list of conns and free it.
+ */
+static int
+destroy_conn(uzfs_mgmt_conn_t *conn)
+{
+	ASSERT(MUTEX_HELD(&conn_list_mtx));
+
+	if (conn->conn_fd >= 0) {
+		if (close_conn(conn) != 0)
```

Comment:
during this close, async_tasks list need to be parsed to remove any async_task with this connection.

Reply:
as we decided to go with a more simple fix yesterday this should not be needed. When async task finishes it checks if conn is still in connection list and if not it drops the reply.